src/main/scala/akka/stream/contrib/DirectoryChanges.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import java.io.File
+import java.nio.file.StandardWatchEventKinds._
+import java.nio.file._
+
+import akka.NotUsed
+import akka.stream.{ ActorAttributes, Attributes, Outlet, SourceShape }
+import akka.stream.scaladsl.Source
+import akka.stream.stage.{ GraphStage, GraphStageLogic, OutHandler, TimerGraphStageLogic }
+import akka.japi.Pair
+import com.sun.nio.file.SensitivityWatchEventModifier
+
+import scala.collection.immutable.Queue
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+object DirectoryChanges {
+  sealed trait Change
+  case object Modified extends Change
+  case object Created extends Change
+  case object Deleted extends Change
+
+  /**
+   * Java API
+   */
+  val modified = Modified
+  /**
+   * Java API
+   */
+  val created = Created
+  /**
+   * Java API
+   */
+  val deleted = Deleted
+
+  /**
+   * Scala API
+   *
+   * Watches a file system directory and streams change events from it.
+   *
+   * @see [[DirectoryChanges]] for details
+   */
+  def apply(directory: Path, pollInterval: FiniteDuration = 500.millis, maxBufferSize: Int = 200): Source[(Path, Change), NotUsed] =
+    Source.fromGraph(new DirectoryChanges(directory, pollInterval, maxBufferSize))
+
+  /**
+   * Java API
+   * @see [[DirectoryChanges]] for details
+   */
+  def forDirectory(directory: Path): Source[Pair[Path, DirectoryChanges.Change], NotUsed] = apply(directory).map(t => Pair.create(t._1, t._2))
```

Comment:
Not sure about these factories, but anyway, if they stay this needs to return javadsl.Source ofc.